builtin/providers/digitalocean/loadbalancer.go

```diff
@@ -0,0 +1,137 @@
+	return func() (interface{}, string, error) {
+		lb, _, err := client.LoadBalancers.Get(loadbalancerId)
+		if err != nil {
+			return nil, "", fmt.Errorf("Error issuing read request in LoadbalancerStateRefreshFunc to DigitalOcean for Load Balancer '%s': %s", loadbalancerId, err)
+		}
+
+		return lb, lb.Status, nil
+	}
+}
+
+func expandStickySessions(config []interface{}) *godo.StickySessions {
+	stickysessionConfig := config[0].(map[string]interface{})
+
+	stickySession := &godo.StickySessions{
+		Type: stickysessionConfig["type"].(string),
+	}
+
+	if v, ok := stickysessionConfig["cookie_name"]; ok {
+		stickySession.CookieName = v.(string)
+	}
+
+	if v, ok := stickysessionConfig["cookie_ttl_seconds"]; ok {
+		stickySession.CookieTtlSeconds = v.(int)
+	}
+
+	return stickySession
+}
+
+func expandHealthCheck(config []interface{}) *godo.HealthCheck {
+	healthcheckConfig := config[0].(map[string]interface{})
+
+	healthcheck := &godo.HealthCheck{
+		Protocol:               healthcheckConfig["protocol"].(string),
+		Port:                   healthcheckConfig["port"].(int),
+		CheckIntervalSeconds:   healthcheckConfig["check_interval_seconds"].(int),
+		ResponseTimeoutSeconds: healthcheckConfig["response_timeout_seconds"].(int),
+		UnhealthyThreshold:     healthcheckConfig["unhealthy_threshold"].(int),
+		HealthyThreshold:       healthcheckConfig["healthy_threshold"].(int),
+	}
+
+	if v, ok := healthcheckConfig["path"]; ok {
+		healthcheck.Path = v.(string)
+	}
+
+	return healthcheck
+}
+
+func expandForwardingRules(config []interface{}) []godo.ForwardingRule {
+	forwardingRules := make([]godo.ForwardingRule, 0, len(config))
+
+	for _, rawRule := range config {
+		rule := rawRule.(map[string]interface{})
+
+		r := godo.ForwardingRule{
+			EntryPort:      rule["entry_port"].(int),
+			EntryProtocol:  rule["entry_protocol"].(string),
+			TargetPort:     rule["target_port"].(int),
+			TargetProtocol: rule["target_protocol"].(string),
+			TlsPassthrough: rule["tls_passthrough"].(bool),
+		}
+
+		if v, ok := rule["certificate_id"]; ok {
+			r.CertificateID = v.(string)
+		}
+
+		forwardingRules = append(forwardingRules, r)
+
+	}
+
+	return forwardingRules
+}
+
+func flattenDropletIds(list []int) []interface{} {
+	vs := make([]interface{}, 0, len(list))
+	for _, v := range list {
+		vs = append(vs, v)
+	}
+	return vs
+}
+
+func flattenHealthChecks(health *godo.HealthCheck) []map[string]interface{} {
+	result := make([]map[string]interface{}, 0, 1)
+
+	r := make(map[string]interface{})
+	r["protocol"] = (*health).Protocol
+	r["port"] = (*health).Port
+	r["path"] = (*health).Path
+	r["check_interval_seconds"] = (*health).CheckIntervalSeconds
+	r["response_timeout_seconds"] = (*health).ResponseTimeoutSeconds
+	r["unhealthy_threshold"] = (*health).UnhealthyThreshold
+	r["healthy_threshold"] = (*health).HealthyThreshold
+
+	result = append(result, r)
+
+	return result
+}
+
+func flattenStickySessions(session *godo.StickySessions) []map[string]interface{} {
```

Comment:
What does DO give us if there are no sticky sessions? I don't know the API, but if `nil` is an option, I worry this and other `flatten`  methods may panic if `nil` is received